src/detect-detection-filter.c

```diff
@@ -440,19 +437,18 @@ static int DetectDetectionFilterTestSig1(void)
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
     alerts += PacketAlertCheck(p, 1);
 
-    if(alerts == 4)
-        result = 1;
+    FAIL_IF_NOT(alerts == 4);
 
     SigGroupCleanup(de_ctx);
     SigCleanSignatures(de_ctx);
```

Comment:
When you use `DetectEngineAppendSig`, you can remove these.

Reply:
Technically unrelated, but please remove anyway :)